### PR TITLE
Chroot command removed

### PR DESCRIPTION
**Pull request by Avatat**
This PR is related to: https://github.com/iocage/iocage/pull/541
I removed `chroot` command and added information about alternative commands (`command`, `execute`).

### CODE REVIEW

**Review by gronke — DISMISSED** [review]
Nice and gentle approach to print a deprecation warning! 😍

File: iocage/lib/iocage.py [hunk 1]
```diff
@@ -386,82 +386,20 @@ def activate(self, zpool):
...
-        if chroot.returncode:
-            ioc_common.logit(
-                {
-                    "level": "WARNING",
-                    "message": "Chroot had a non-zero exit code!"
-                },
-                _callback=self.callback,
-                silent=self.silent)
-
+        """Depracated: Chroots into a jail and runs a command, or the shell."""
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: `deprecated`

File: iocage/lib/iocage.py [hunk 2]
```diff
@@ -386,82 +386,20 @@ def activate(self, zpool):
...
+        ioc_common.logit(
+            {
+                "level": "INFO",
+                "message": 
+                ("iocage chroot is depracated. If you need execute shell inside"
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: Spelling here as well :) In addition, `to execute a`

File: iocage/lib/iocage.py [hunk 3]
```diff
@@ -386,82 +386,20 @@ def activate(self, zpool):
...
+                 " jail use: iocage console" if len(command) == 0 else
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: ` the jail`

File: iocage/lib/iocage.py [hunk 4]
```diff
@@ -386,82 +386,20 @@ def activate(self, zpool):
...
-                silent=self.silent)
-
+        """Depracated: Chroots into a jail and runs a command, or the shell."""
+        ioc_common.logit(
+            {
+                "level": "INFO",
+                "message": 
+                ("iocage chroot is depracated. If you need execute shell inside"
+                 " jail use: iocage console" if len(command) == 0 else 
+                 "iocage chroot is depracated. If you need execute command"
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: Spelling :) `to execute a`

File: iocage/lib/iocage.py [hunk 5]
```diff
@@ -386,82 +386,20 @@ def activate(self, zpool):
...
+                 " inside jail use: iocage execute")
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: ` inside the jail` and it's just `iocage exec`

**Review by skarekrow — CHANGES_REQUESTED** [review]
Looks good, some changes needed and a travis failure :+1:

**Comment by gronke** [comment]
Travis still has a few complaints:

```
./iocage/lib/iocage.py:393:27: W291 trailing whitespace
./iocage/lib/iocage.py:394:80: E501 line too long (80 > 79 characters)
./iocage/lib/iocage.py:395:71: W291 trailing whitespace
./iocage/lib/iocage.py:402:1: W293 blank line contains whitespace
```

**Comment by skarekrow** [comment]
I should also add that we need to replace the instances of `chroot` within the program as well, `lib/ioc_upgrade.py` and such.

**Review by skarekrow — DISMISSED** [review]

**Comment by Avatat** [comment]
@gronke, @skarekrow, thank you for kind words and help :)
Everything was fixed.

File: iocage/lib/iocage.py [hunk 6]
```diff
@@ -386,81 +386,19 @@ def activate(self, zpool):
...
-                silent=self.silent)
+        """Deprecated: Chroots into a jail and runs a command, or the shell."""
+        ioc_common.logit(
+            {
+                "level": "INFO",
+                "message":
+                ("iocage chroot is deprecated. If you need to execute a shell"
+                 " inside the jail use: iocage console" if len(command) == 0
+                 else "iocage chroot is deprecated. If you need to execute a"
+                 " command inside the jail use: iocage exec")
```
Comment (gronke) [inline iocage/lib/iocage.py]: Suggestion to improve the readability of the above text block:

```python
(
    "iocage is deprecated. "
    "If you need to execute a {} inside the jail use: {}"
).format(*[
  ["shell", "iocage console"],
  ["command", "iocage exec"]
][int(len(command) == 0)])
```

**Review by gronke — DISMISSED** [review]

File: iocage/lib/iocage.py [hunk 7]
```diff
@@ -391,10 +391,13 @@ def chroot(self, command):
             {
                 "level": "INFO",
                 "message":
-                ("iocage chroot is deprecated. If you need to execute a shell"
-                 " inside the jail use: iocage console" if len(command) == 0
-                 else "iocage chroot is deprecated. If you need to execute a"
-                 " command inside the jail use: iocage exec")
+                (
+                    "iocage is deprecated. "
```
Comment (skarekrow) [inline iocage/lib/iocage.py]: `chroot` not iocage :D

**Review by skarekrow — CHANGES_REQUESTED** [review]

File: iocage/lib/iocage.py (hunk 7 above)
Comment (Avatat) [reply iocage/lib/iocage.py]: Of course! :D Fixed.

I will update `lib/ioc_upgrade.py` soon, in few hours.

**Review by skarekrow — APPROVED** [review]

**Comment by skarekrow** [comment]
Thanks for all the work @Avatat :+1:

**Comment by reelsense** [comment]
Is this related?

https://github.com/iocage/iocage/issues/535

---
x-post: https://github.com/iocage/iocage/pull/545

https://github.com/iocage/iocage/pull/542

**Pull request merged**